Tools/autotest/pysim/util.py

```diff
@@ -190,7 +190,15 @@ def __init__(self, binary, valgrind=False, gdb=False, wipe=False, synthetic_cloc
         pexpect_autoclose(self)
         # give time for parameters to properly setup
         time.sleep(3)
-        self.expect('Waiting for connection',timeout=300)
+        if gdb:
+            # if we run GDB we do so in an xterm.  "Waiting for
+            # connection" is never going to appear on xterm's output.
+            #... so let's give it another magic second.
+            time.sleep(1)
```

Comment:
I  really hate magic sleeps, but it seems there's already a "TODO" below to fix it